tests/test-dt.cpp

```diff
@@ -46,51 +46,65 @@ TEST_P(DTTests, DatatypeDecl)
     if (sc.is_logging_solver) {
       return;
     }
-    Term five = s->make_term(5, intsort);
+    /* Original location of five declaration
+    if (s->get_solver_enum() != GENERIC_SOLVER) {
+      Term five = s->make_term(5, intsort);
+    }
+    */
 
     // Make datatype sort
     DatatypeDecl consListSpec = s->make_datatype_decl("list");
 
     DatatypeConstructorDecl nildecl = s->make_datatype_constructor_decl("nil");
     DatatypeConstructorDecl consdecl = s->make_datatype_constructor_decl("cons");
     s->add_selector(consdecl, "head", s->make_sort(INT));
```

Comment:
plz add somewhere in this test a comparison between datatype constructors -- 
1. two that are the same (and assert that they are the same) 
2. two that are different (and assert that they are different)

for assertions use something like `ASSERT_TRUE` or `ASSERT_FALSE` or `ASSERT_EQ` or `ASSERT_EQUAL`.